core/scss/components/list-item/mixins.scss

```diff
@@ -151,6 +152,28 @@
     }
   }
 
+  &:hover:not(.disabled),
```

Comment:
The issue with this added logic is that ListItemHeader automatically applies the read-only class.  We just need a :focus and :hover style applied.